File: src/transformers/features/validators/CreateAssertTransformer.ts
import { AssertProgrammer } from "../../../programmers/AssertProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateAssertTransformer {
    export const transform = (equals: boolean) =>
        GenericTransformer.factory(
            equals ? "createAssertEquals" : "createAssert",
        )((project, modulo) =>
            AssertProgrammer.generate(project, modulo, equals),
        );
}

File: packages/transform/src/features/CreateAssertTransformer.go
package features

import (
	"github.com/samchon/typia/packages/core/src/programmers"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var CreateAssertTransformer = createAssertTransformerNamespace{}

type createAssertTransformerNamespace struct{}

func (createAssertTransformerNamespace) Transform(equals bool, guard bool) internal.Task {
	return internal.GenericTransformer.Factory(programmers.AssertProgrammer.Write)
}
